whisper/whisperv5/filter_test.go

```diff
@@ -796,19 +796,52 @@ func TestVariableTopics(t *testing.T) {
 	}
 
 	for i := 0; i < 4; i++ {
-		arr := make([]byte, i+1, 4)
-		copy(arr, env.Topic[:i+1])
-
-		f.Topics[4] = arr
+		env.Topic = BytesToTopic(f.Topics[i])
 		match = f.MatchEnvelope(env)
 		if !match {
 			t.Fatalf("failed MatchEnvelope symmetric with seed %d, step %d.", seed, i)
 		}
 
-		f.Topics[4][i]++
+		f.Topics[i][1]++
```

Comment:
Why not `f.Topics[i][4]++` ?

Reply:
TopicLength is 4. So f.Topics[i][4] is out of range. 
I've added local const for this and changed value to 3(last byte).